os/fs/driver/mtd/smart.c

```diff
@@ -4070,12 +4070,19 @@ static int smart_validate_crc(FAR struct smart_struct_s *dev)
 	if (crc != *((uint32_t *)header->crc32)) {
 		return -EIO;
 	}
-#else
+#elif defined(CONFIG_SMART_CRC_8)
 	/* Test 8-bit CRC for CRC8 & basic case for smart_scan. */
 	if (crc != header->crc8 && crc != SMART_ERASEDSTATE_CRC) {
 		return -EIO;
 	}
 
+#else
+	/*CRC not enabled, check only for first three bytes i.e. logicalsector and sequence number. */
+
+	if (crc != header->crc8 && (header->logicalsector[0] != 0xFF || header->logicalsector[1] != 0xFF || header->seq != 0xFF)) {
```

Comment:
Don't we need to change "0xFF" into 'SMART_ERASEDSTATE'?

Reply:
Agreed, thanks for the suggestion. I'll make the changes.